extensions/amp-ad/0.1/test/test-amp-ad-api-handler.js

```diff
@@ -65,17 +68,80 @@ describe('amp-ad-api-handler', () => {
       }));
     });
 
-    it('should resolve startUp() when render-start API is called', () => {
-      expect(iframe.style.visibility).to.equal('hidden');
-      iframe.postMessageToParent({
-        sentinel: 'amp3ptest' + testIndex,
-        type: 'render-start',
+    describe('should resolve startUp', () => {
+      it('if render-start NOT supported', () => {
+        expect(iframe.style.visibility).to.equal('hidden');
+        iframe.postMessageToParent({
+          sentinel: 'amp3ptest' + testIndex,
+          type: 'bootstrap-loaded',
+        });
+        return startUpPromise.then(() => {
+          expect(iframe.style.visibility).to.equal('');
+        });
       });
-      return startUpPromise.then(() => {
-        expect(iframe.style.visibility).to.equal('');
+
+      it('if render-start supported, receive render-start first', () => {
+        adImpl.adType = 'doubleclick';
+        apiHandler = new AmpAdApiHandler(adImpl, adImpl.element);
+        apiHandler.noContentCallback_ = () => {};
+        const beforeAttachedToDom = element => {
+          element.setAttribute('data-amp-3p-sentinel', 'amp3ptest' + testIndex);
+          startUpPromise = apiHandler.startUp(element, true);
+        };
+        const noContentCallbackSpy
+            = sandbox.spy(apiHandler, 'noContentCallback_');
+        return createIframeWithMessageStub(window, beforeAttachedToDom)
+            .then(newIframe => {
+              iframe = newIframe;
+              expect(iframe.style.visibility).to.equal('hidden');
+              iframe.postMessageToParent({
+                sentinel: 'amp3ptest' + testIndex,
+                type: 'render-start',
+              });
+              return startUpPromise.then(() => {
+                expect(iframe.style.visibility).to.equal('');
+              }).then(() => {
+                iframe.postMessageToParent({
+                  sentinel: 'amp3ptest' + testIndex,
+                  type: 'no-content',
+                });
+                return expectPostMessage(iframe.contentWindow, window, {
+                  sentinel: 'amp3ptest' + testIndex,
+                  type: 'no-content',
+                }).then(() => {
+                  expect(noContentCallbackSpy).to.not.been.called;
+                });
+              });
+            });
+      });
+
```

Comment:
TODO: add another test that ad should not respond to `bootstrap-loaded` when `render-start` is supported.